src/start/hello-xdp.md

```diff
@@ -67,54 +68,71 @@ Simple!
-Fortunately, we have a program ready in `myapp/src/main.rs` which is going to do that for us.
+Fortunately, we have a generated program ready in `myapp/src/main.rs` which is going to do that for us.
 
 ### Starting Out
 
 The generated application has the following content:
 
 ```rust,ignore
-{{#rustdoc_include ../../examples/myapp-00/myapp/src/main.rs }}
+{{#rustdoc_include ../../examples/myapp-01/myapp/src/main.rs:all }}
 ```
 
-Let's adapt it to load our program.
+Let's look at the details of this program.
 
-We will add a dependency on `ctrlc` to `myapp/Cargo.toml`:
+There is a dependency on `ctrlc` added to `myapp/Cargo.toml`:
 ```toml
 {{#include ../../examples/myapp-01/myapp/Cargo.toml:11}}
 ```
 
-Add the following declarations at the top of the `myapp/src/main.rs`:
+The following `use` declarations are added at the top of the `myapp/src/main.rs`:
 
 ```rust,ignore
 {{#rustdoc_include ../../examples/myapp-01/myapp/src/main.rs:use }}
 ```
 
-Then we'll adapt the `try_main` function to load our program:
+
+`try_main` function loads our program:
 
 ```rust,ignore
 {{#rustdoc_include ../../examples/myapp-01/myapp/src/main.rs:try_main }}
 ```
 
-The program takes two positional arguments
-- The path to our eBPF application
-- The interface we wish to attach it to (defaults to `eth0`)
 
-The line `let mut bpf = Bpf::load_file(&path)?;`:
-- Opens the file
+The program optionally takes an argument for the interface we wish to attach it to (defaults to `eth0`).
+
+`include_bytes_aligned!` copies the contents of the BPF object file
+into a variable at the compile time.
+The statement `let mut bpf = Bpf::load_file(data)?;`:
 - Reads the ELF contents
 - Creates any maps
 - If your system supports BPF Type Format (BTF), it will read the current BTF description and performs any necessary relocations
 
-Once our file is loaded, we can extract the XDP probe with `let probe: &mut Xdp = bpf.program_mut("xdp")?.try_into()?;` and then load it in to the kernel with `probe.load()`.
+Once our file is loaded, we can extract the XDP program with `let program: &mut Xdp = bpf.program_mut("myapp")?.try_into()?;` and then load it in to the kernel with `program.load()`.
 
-Finally, we can attach it to an interface with `probe.attach(&iface, XdpFlags::default())?;`
+Finally, we can attach it to an interface with `program.attach(&opt.iface, XdpFlags::default())?;`
 
 Let's try it out!
 
 ```console
+# change directory to the root or myapp
 $ cargo build
-$ sudo ./target/debug/myapp --path ./target/bpfel-unknown-none/debug/myapp wlp2s0
+$ cargo xtask run -- -h
+myapp 0.1.0
+
+USAGE:
+    myapp [OPTIONS]
+
+FLAGS:
+    -h, --help       Prints help information
+    -V, --version    Prints version information
+
+OPTIONS:
+    -i, --iface <iface>     [default: eth0]
+
+# add -i or --iface parameter if need to attach to an interface other than eth0
+# - for example, cargo xtask run -- -h -- -i wlp2s0
```

Comment:
This should be `cargo xtask run -- -i wlp2s0`

Reply:
fixed in 425f4b0